Optional custom kill commands

philpax opened the pull request:
I run ComfyUI through Docker, which responds poorly (i.e. not at all) to the kill signal that LMP normally uses. To work around this, I've added support for optionally specifying custom kill commands, which will be executed before the primary termination mechanism.

This is semi-consistent with systemd's [`ExecStop`](https://www.freedesktop.org/software/systemd/man/latest/systemd.service.html#ExecStop=), which is what I was using prior to LMP. I chose to terminate the process regardless of the stop command, unlike systemd, to ensure that the rest of the invariants that LMP relies upon remain in place (i.e. the PID is dead-and-gone when the function exits).

perk11 commented:
This make sense. I think the part where it terminates the process regardless can be reworked in the future to be made optional. 

Would you mind adding a test for this functionality? Both for the kill command and for the config file. It will be good to merge once the tests are added.

philpax commented:
I've added tests for both, but I'm not sure about either one:
- the config test doesn't really check much at all, especially because it doesn't go through the parser. I'm not sure what else I could test here, and I suspect the tests I've added could be removed with no real loss.
- the kill test uses a temporary file to determine whether it's been run. This works fine, but it's not the most elegant thing.

perk11 commented:
Thank you, everything looks great, I'm only going to simplify killCommand() test a little bit

perk11 commented:
https://github.com/perk11/large-model-proxy/pull/36/commits/13d19f682fa518f949377a4df9345f911387489b this is what I did to the kill-test for reference

The other scenarios are already covered in a different test, trying to keep this new to minimum necessary code :)
